src/classes/HH_OppContactRoles_TDTM.cls

```diff
@@ -55,8 +55,20 @@ public class HH_OppContactRoles_TDTM extends TDTM_Runnable {
         if (!currentHouseholdsSettings.npo02__Household_Contact_Roles_On__c)
             return null;
 
-        list<Opportunity> opportunities = newlist;
-        list<Opportunity> oldOpportunities = oldlist;
+        List<Opportunity> opportunities = new List<Opportunity>();
+        List<Opportunity> oldOpportunities = new List<Opportunity>();
+        for (Integer i = 0, size = newList.size(); i < size; i++) {
+            Opportunity opp = (Opportunity) newList[i];
```

Comment:
Not sure if you want to change this here but should we grab the old contact by id vs index?

Reply:
In this example it is fine since `newList` is not retrieved again in the method, and `newList` and `oldList` are in sync to start with.

Reply:
Gotcha! I was thinking our preference was to always grab the old contact using the id vs relying on the order being the same. Thanks for clarifying.